libcodechecker/log/log_parser.py

```diff
@@ -0,0 +1,791 @@
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+
+            _, err = proc.communicate("")
+            return err
+        except OSError as oerr:
+            LOG.error("Error during process execution: " + cmd + '\n' +
+                      oerr.strerror + "\n")
+
+    def __parse_compiler_includes(self, lines):
+        """
+        Parse the compiler include paths from a string
+        """
+        start_mark = "#include <...> search starts here:"
+        end_mark = "End of search list."
+
+        include_paths = []
+
+        do_append = False
+        for line in lines.splitlines(True):
+            line = line.strip()
+            if line.startswith(end_mark):
+                break
+            if do_append:
+                # On OSX there are framework includes,
+                # where we need to strip the "(framework directory)" string.
+                # For instance:
+                # /System/Library/Frameworks (framework directory)
+                fpos = line.find("(framework directory)")
+                if fpos == -1:
+                    include_paths.append(line)
+                else:
+                    include_paths.append(line[0:fpos-1])
+
+            if line.startswith(start_mark):
+                do_append = True
+
+        return include_paths
+
+    def __filter_compiler_includes(self, include_dirs):
+        """
+        Filter the list of compiler includes.
+        We want to elide GCC's include-fixed and instrinsic directory.
+        See docs/gcc_incompatibilities.md
+        """
+
+        def contains_intrinsic_headers(include_dir):
+            """
+            Returns True if the given directory contains at least one intrinsic
+            header.
+            """
+            if not os.path.exists(include_dir):
+                return False
+            for f in os.listdir(include_dir):
+                if f.endswith("intrin.h"):
+                    return True
+            return False
+
+        result = []
+        for include_dir in include_dirs:
+            # Skip GCC's fixinclude dir
+            if os.path.basename(
+                    os.path.normpath(include_dir)) == "include-fixed":
+                continue
+            if contains_intrinsic_headers(include_dir):
+                continue
+            result.append(include_dir)
+        return result
+
+    def get_compiler_includes(self, action):
+        """
+        Returns a list of default includes of the given compiler.
+
+        action -- A BuildAction object containing the details of compilation.
+        """
+        # The first sysroot flag found among the compilation options is added
+        # to the command below to give a more precise default include list.
+        # Absence of any sysroot flags results in an empty string.
+        sysroot = ""
+        for i, item in enumerate(action.analyzer_options):
```

Comment:
Should we check how the compiler works if multiple sysroot is passed? Which one takes precedence.

Reply:
The last one will take precedence, so I'll fix it.